### PR TITLE
Fix for #8 issue

### PR DESCRIPTION
**Pull request by AmirMehrabi**
If the '.env.example' didn't exist, the commands would fail.

Now we check the environment list, loop through it, and if any of the files are not present we'll create them with a sample ".env.example".

### CODE REVIEW

**Comment by AmirMehrabi** [comment]
The reported issue: #8

**Review by a-bashtannik — CHANGES_REQUESTED** [review]
The default `.env.example` file differs for [Larave v8.x](https://raw.githubusercontent.com/laravel/laravel/8.x/.env.example) and [Laravel v9.x ](https://raw.githubusercontent.com/laravel/laravel/9.x/.env.example).

**Comment by AmirMehrabi** [comment]
Will fix this in half an hour. @a-bashtannik

File: src/Support/LaravelFinder.php [hunk 1]
```diff
@@ -47,7 +49,12 @@ public function environmentFilePaths(): array
     {
         foreach($this->environmentFiles as $envFile) {
             if(!file_exists( $envFile )) {
-                File::copy(base_path() . "/vendor/worksome/envy/config/.env.example", $envFile);
+                $fileName = substr($envFile, strrpos($envFile, '/') + 1);
```
Comment (a-bashtannik) [inline src/Support/LaravelFinder.php:52]: `basename($envFile)`

File: src/Support/LaravelFinder.php [hunk 2]
```diff
@@ -47,7 +49,12 @@ public function environmentFilePaths(): array
...
+                if ($fileName == '.env.example') {
+                    File::copy(base_path() . "/vendor/worksome/envy/config/$fileName.".substr(app()->version(),0,1), $envFile);
+                } else {
+                    File::copy(base_path() . "/vendor/worksome/envy/config/$fileName", $envFile);
```
Comment (a-bashtannik) [inline src/Support/LaravelFinder.php:56]: Will raise the "failed to open stream" warning if the file does not exist. Unpredictable behavior.

File: src/Support/LaravelFinder.php [hunk 3]
```diff
@@ -47,7 +49,12 @@ public function environmentFilePaths(): array
     {
         foreach($this->environmentFiles as $envFile) {
             if(!file_exists( $envFile )) {
-                File::copy(base_path() . "/vendor/worksome/envy/config/.env.example", $envFile);
+                $fileName = substr($envFile, strrpos($envFile, '/') + 1);
+                if ($fileName == '.env.example') {
+                    File::copy(base_path() . "/vendor/worksome/envy/config/$fileName.".substr(app()->version(),0,1), $envFile);
```
Comment (a-bashtannik) [inline src/Support/LaravelFinder.php:54]: Better use `version_compare()`

**Comment by AmirMehrabi** [comment]
@a-bashtannik Thanks for pointing me in the right direction. 

I didn't pay attention to the fact that we're supporting two versions of Laravel.

So the way I'm handling it now is that we check the laravel version first, based on the version we copy the ".env.example" file accordingly.

**Review by a-bashtannik — CHANGES_REQUESTED** [review]

File: src/Support/LaravelFinder.php (hunk 1 above)
Comment (AmirMehrabi) [reply src/Support/LaravelFinder.php:52]: Done

File: src/Support/LaravelFinder.php (hunk 2 above)
Comment (AmirMehrabi) [reply src/Support/LaravelFinder.php:56]: You're right. But the goal here was to resolve the issue for the main env file (.env.example), which this PR will do that. 
What's your suggestion to the issue? How do you think we can get rid of this issue entirely for all the files?

**Pull request closed without merging**